File: 1st/Hyeseung/Programmers/등수 매기기.java
class Solution {
    public int[] solution(int[][] score) {
        int[] answer = new int[score.length];
        double[] dob = new double[score.length];
        
        for(int i = 0; i < score.length; i++)
            dob[i] = (score[i][0] + score[i][1]) / 2.0;
        
        for(int i = 0; i < answer.length; i++) {
            int rank = 1;
            for(int j = 0; j < answer.length; j++) if(dob[i] < dob[j]) rank++;
            answer[i] = rank;
        }

        return answer;
    }
}

File: 3st/Hyeonseo/programmer/등수 매기기.py
def solution(score):
    n = len(score)
    answer = [1] * n
    score_sum = [sum(s) for s in score]  
    for i in range(n):
        for j in range(n):
            if score_sum[i] < score_sum[j]:
                answer[i] += 1
    return answer
